### ISSUE TITLE
Add SAML / 3rd Party Authentication Support

### ISSUE THREAD

**Issue by naodwalk** [hjdhjd/homebridge-myq#1064]
**Is your feature request related to a problem? Please describe:**
There is no way to authenticate to myQ with this plugin if a user registers with their Apple, Google or Amazon account. 

**Describe the solution you'd like:**
Allow users the ability to authenticate to the API using their SAML credentials, or at least warn the users in the README that this is not supported.

**Describe alternatives you've considered:**
Delete my SAML account and recreate one using myQ user/password authentication

**Additional context:**
If you're using a custom API, then I understand this could be a limitation with that. However, if you wrote your own, I'd be happy to Wireshark my phone to try and see where the authentication happens and what the request/response looks like.

**Comment by hjdhjd** [comment]
I have no intentions currently to support non-myQ accounts. There are several technical considerations that just aren't worth the effort, though if someone wants to take it on, I'm open to accepting a PR.

TL;DR: If you want to use this plugin, you're going to need a myQ account.

If someone wants to work on supporting other login types, I'm all for it as long as it's well-implemented...but I'm focusing my time elsewhere given the solve is pretty straightforward - use a myQ account. 😄
